infrastructure.md

```diff
@@ -0,0 +1,54 @@
+# CPG Infrastructure
+
+## Dataset
+
+A dataset is one defined collection of resources. It has:
+
+- A GCP project
+- A set of buckets
+- A set of secrets to manage access (access_group_cache, cromwell access keys)
+- A set of allowed_repositories
+
+## Components
```

Comment:
Cromwell might be worth mentioning here, too, particularly since long term we'd like to move to Terra for that functionality.